src/components/Form/Form.tsx

```diff
@@ -1,62 +1,153 @@
-    )?.value;
-    const rateAIIntelligence = rangeValue;
+  const onWhereUsedChange = (
+    event: SelectChangeEvent<typeof whereAIIsUsed>
+  ) => {
+    const {
+      target: { value },
+    } = event;
+    const newOption = typeof value === "string" ? value.split(",") : value;
+    setWhereAIIsUsed(newOption);
+    setAI({ ...Ai, whereAIIsUsed: newOption });
+  };
+
+  const onTypeChange = (e: SelectChangeEvent) => {
+    setTypeOfAI(e.target.value);
+    setAI({ ...Ai, TypeOfAI: e.target.value });
+  };
 
+  const onRateSliderChange = (_: Event, newValue: number | number[]) => {
+    const value = Array.isArray(newValue) ? newValue[0] : newValue;
+    setAI({ ...Ai, rateAIIntelligence: value });
+    setRangeValue(value);
+  };
+
+  const onRateInputChange = (event: React.ChangeEvent<HTMLInputElement>) => {
+    const value = event.target.value === "" ? 0 : Number(event.target.value);
+    setAI({ ...Ai, rateAIIntelligence: value });
+    setRangeValue(value);
+  };
+
+  const handleBlur = () => {
+    if (rangeValue < 0) {
+      setRangeValue(0);
+    } else if (rangeValue > 100) {
+      setRangeValue(100);
+    }
+  };
+  const validateAI = (Ai: AI) => {
     if (
-      levelOfAI.length === 0 ||
-      whereAIIsUsed.length === 0 ||
-      !TypeOfAI ||
-      rateAIIntelligence === 0
+      Ai.levelOfAI.length === 0 ||
+      Ai.whereAIIsUsed.length === 0 ||
+      !Ai.TypeOfAI ||
+      Ai.rateAIIntelligence === 0
     ) {
-      setErrorMessage("Please select at least one option.");
-      return;
+      console.log(Ai);
+      return "Please fill all the fields";
     }
+  };
 
-    setFormData((prevData) => [
-      ...prevData,
-      {
-        levelOfAI,
-        whereAIIsUsed,
-        TypeOfAI,
-        rateAIIntelligence,
-      },
-    ]);
+  const handleSubmit = (e: React.FormEvent) => {
+    e.preventDefault();
+    setErrorMessage("");
+
+    const error = validateAI(Ai);
+    if (error) {
+      setErrorMessage(error);
+      return;
+    }
+    onSubmit(Ai);
+    console.log(Ai);
   };
   const handleClear = () => {
-    document
-      .querySelectorAll('input[name="level"]')
-      .forEach((input) => ((input as HTMLInputElement).checked = false));
-    document
-      .querySelectorAll('input[name="occupancy"]')
-      .forEach((input) => ((input as HTMLInputElement).checked = false));
-    (
-      document.querySelector('input[name="type"]:checked') as HTMLInputElement
-    ).checked = false;
+    setErrorMessage(null);
     setRangeValue(0);
+    setAI(defaultFormState);
+    setLevelOfAI([]);
+    setWhereAIIsUsed([]);
+    setTypeOfAI("");
+    console.log(Ai);
+    document.querySelectorAll('input[type="checkbox"]').forEach((element) => {
+      const checkbox = element as HTMLInputElement;
+      checkbox.checked = false;
+    });
+    document.querySelectorAll('input[type="radio"]').forEach((element) => {
+      const radio = element as HTMLInputElement;
+      radio.checked = false;
+    });
```

Comment:
try to avoid thinks like `document.querySelectorAll` as much as possible

Reply:
Actually, i tried to do something similar to Adrian example, but i had again issue with clear button (it's seems even in that example was this issue). So I wrote this piece of code to fix clearing

Reply:
to tell the truth could get this maybe we should  discuss this separately